cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowUtil.qll

```diff
@@ -422,8 +422,18 @@ private module ThisFlow {
  * Holds if data flows from `nodeFrom` to `nodeTo` in exactly one local
  * (intra-procedural) step.
  */
-cached
 predicate localFlowStep(Node nodeFrom, Node nodeTo) {
+  simpleLocalFlowStep(nodeFrom, nodeTo)
+}
+
+/**
+ * INTERNAL: do not use.
+ *
+ * This is the local flow predicate that's used as a building block in global
+ * data flow. It may have less flow than the `localFlowStep` predicate.
+ */
+cached
+predicate simpleLocalFlowStep(Node nodeFrom, Node nodeTo) {
```

Comment:
If you move this to the `Private` module you can avoid exposing it.

Reply:
That's correct, but it's not trivial to move it. Its dependencies would also have to move, and it might interfere with caching and require a cached module. I agree that the C# code organisation is better, but I'd rather reorganize the C++ code in a C++-only PR.